service/src/main/java/com/ge/predix/acs/privilege/management/dao/GraphGenericRepository.java

```diff
@@ -447,6 +448,20 @@ private void checkTraversalLimitOrFail(final Set<Attribute> attributes) {
         return parentSet;
     }
 
+    public Set<String> getEntityAndDescendantsIds(final E entity) {
+        if (entity == null) {
+            return Collections.emptySet();
+        }
+
+        Set<String> descendantsIds = new HashSet<>();
+
+        this.graph.traversal().V(entity.getId()).has(getEntityIdKey()).emit().repeat(in().has(getEntityIdKey()))
+        .until(eq(null)).values(getEntityIdKey()).toStream().forEach(id -> {
+            descendantsIds.add((String) id);
```

Comment:
Rather than use forEach to add each item to the result set. You could use Java 8 [collectors](https://docs.oracle.com/javase/8/docs/api/java/util/stream/Collectors.html). E.g.  
`.toStream().collect(Collectors.toSet()).`